build.cmd

```diff
@@ -20,17 +25,21 @@ set __ThisScriptFull="%~f0"
 :: toolset if it is installed. Finally, we will fail the script if no supported VS instance
 :: can be found.
 
-if defined VisualStudioVersion goto :Run
+if defined VisualStudioVersion (
+    if not defined __VSVersion echo %__MsgPrefix%Detected Visual Studio %VisualStudioVersion% developer command ^prompt environment
+    goto :Run
+) 
 
+echo %__MsgPrefix%"Searching for Visual Studio 2017 or 2015 installation"
 set _VSWHERE="%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
 if exist %_VSWHERE% (
-  for /f "usebackq tokens=*" %%i in (`%_VSWHERE% -latest -property installationPath`) do set _VSCOMNTOOLS=%%i\Common7\Tools
+for /f "usebackq tokens=*" %%i in (`%_VSWHERE% -latest -prerelease -property installationPath`) do set _VSCOMNTOOLS=%%i\Common7\Tools
```

Comment:
This check is duplicated in several places currently. Add `-prerelease` to all of them?

Reply:
I agree that it is best to get it in to be consistent everywhere - anyway this is the problem which is created by code duplicates